pandas/core/series.py

```diff
@@ -657,12 +657,12 @@ def __getitem__(self, key):
         try:
             result = self.index.get_value(self, key)
 
-            if not is_scalar(result):
+            if not is_scalar(result) and key in self.index:
```

Comment:
why is this check necessary?

Reply:
It will cause that some tests fail,  `if not is_scalar(self.index.get_loc(key))`  will raise ValueError when type of key is Int not String. So by using [0] is another recommended way to get the first item with a String Index in Series? I will try to use `try...except...` to resolve it , Is this way good or a recommended way?      
 Or I should adjust three failed tests?        
* `test_applymap` in `pandas.tests.frame.test_apply.TestDataFrameApply.py`,    
* `test_match_findall_flags` in `pandas.tests.test_strings.TestStringMethods.py`,  
* `test_constructor_from_items` in `pandas.tests.frame.test_constructors.TestDataFrameConstructors.py`       
I'm poor in programing skills and time, sorry about that {:-()}.